src/Concerns/AddsFieldsToQuery.php

```
@@ -73,41 +85,95 @@ protected function addRequestedModelFieldsToQuery(): void
+                $possibleRelatedNames[] = Str::camel($tableName);
+            } elseif ($strategy === 'none') {
+                $possibleRelatedNames[] = $tableName;
+            }
+        }
+
+        // Get fields with potential snake_case conversion
+        $fields = $this->request->fields();
+
+        if (config('query-builder.convert_field_names_to_snake_case', false)) {
+            $fields = $fields->mapWithKeys(fn ($fields, $table) => [
+                $table => collect($fields)->map(fn ($field) => Str::snake($field)),
```

PHPMD (2.15.0)

[error] Avoid using static access to class '\Illuminate\Support\Str' in method 'getRequestedFieldsForRelatedTable'. (undefined)

(StaticAccess)